From 9a25191baf379462ea2d6c508f5a5e20b4d7fc83 Mon Sep 17 00:00:00 2001
From: Greg Kroah-Hartman <gregkh@suse.de>
Date: Tue, 24 Feb 2009 20:11:39 -0800
Subject: Staging: p9auth: remove unneeded header file

From: Greg Kroah-Hartman <gregkh@suse.de>

The p9auth.h file is not needed, move the stuff into p9auth.c file and
delete it.

Cc: Ashwin Ganti <ashwin.ganti@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/p9auth/p9auth.c |   33 ++++++++++++++++++++++++++++++++-
 drivers/staging/p9auth/p9auth.h |   35 -----------------------------------
 2 files changed, 32 insertions(+), 36 deletions(-)

--- a/drivers/staging/p9auth/p9auth.c
+++ b/drivers/staging/p9auth/p9auth.c
@@ -33,7 +33,38 @@
 #include <linux/crypto.h>
 #include <linux/sched.h>
 #include <linux/cred.h>
-#include "p9auth.h"
+
+#ifndef CAP_MAJOR
+#define CAP_MAJOR 0
+#endif
+
+#ifndef CAP_NR_DEVS
+#define CAP_NR_DEVS 2		/* caphash and capuse */
+#endif
+
+#ifndef CAP_NODE_SIZE
+#define CAP_NODE_SIZE 20
+#endif
+
+#define MAX_DIGEST_SIZE  20
+
+struct cap_node {
+	char data[CAP_NODE_SIZE];
+	struct list_head list;
+};
+
+struct cap_dev {
+	struct cap_node *head;
+	int node_size;
+	unsigned long size;
+	struct semaphore sem;
+	struct cdev cdev;
+};
+
+int cap_trim(struct cap_dev *);
+ssize_t cap_write(struct file *, const char __user *, size_t, loff_t *);
+char *cap_hash(char *plain_text, unsigned int plain_text_size, char *key, unsigned int key_size);
+void hex_dump(unsigned char * buf, unsigned int len);
 
 int cap_major = CAP_MAJOR;
 int cap_minor = 0;
--- a/drivers/staging/p9auth/p9auth.h
+++ /dev/null
@@ -1,35 +0,0 @@
-#ifndef CAP_MAJOR
-#define CAP_MAJOR 0
-#endif
-
-#ifndef CAP_NR_DEVS
-#define CAP_NR_DEVS 2		/* caphash and capuse */
-#endif
-
-#ifndef CAP_NODE_SIZE
-#define CAP_NODE_SIZE 20
-#endif
-
-#define MAX_DIGEST_SIZE  20
-
-struct cap_node {
-	char data[CAP_NODE_SIZE];
-	struct list_head list;
-};
-
-struct cap_dev {
-	struct cap_node *head;
-	int node_size;
-	unsigned long size;
-	struct semaphore sem;
-	struct cdev cdev;
-};
-
-extern int cap_major;
-extern int cap_nr_devs;
-extern int cap_node_size;
-
-int cap_trim(struct cap_dev *);
-ssize_t cap_write(struct file *, const char __user *, size_t, loff_t *);
-char *cap_hash(char *plain_text, unsigned int plain_text_size, char *key, unsigned int key_size);
-void hex_dump(unsigned char * buf, unsigned int len);
